src/Microsoft.AspNetCore.OData/Formatter/ODataInputFormatter.cs

```diff
@@ -142,6 +142,7 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
                     }
 
                     logger.LogError(ex, String.Empty);
+                    throw ex;
```

Comment:
Throwing these exceptions is probably going to break some amount of callers. Can we add a behavior flag for this so that we default to the current behavior, and it can be toggled to get the exceptions?